From 7d3d85483a6c4085de5c016b86838681e97e6577 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Fri, 20 May 2016 11:31:17 +0800
Subject: [PATCH 35/93] armv8: ls1012a: enable two esdhc host controllers
 support

LS1012A chip has two esdhc host controllers, and this patch
is to enable two controllers support for it.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
---
 .../include/asm/arch-fsl-layerscape/immap_lsch2.h  |    1 +
 include/configs/ls1012a_common.h                   |    2 ++
 2 files changed, 3 insertions(+)

diff --git a/arch/arm/include/asm/arch-fsl-layerscape/immap_lsch2.h b/arch/arm/include/asm/arch-fsl-layerscape/immap_lsch2.h
index 24add1a..6918757 100644
--- a/arch/arm/include/asm/arch-fsl-layerscape/immap_lsch2.h
+++ b/arch/arm/include/asm/arch-fsl-layerscape/immap_lsch2.h
@@ -19,6 +19,7 @@
 #define CONFIG_SYS_GIC400_ADDR			(CONFIG_SYS_IMMR + 0x00400000)
 #define CONFIG_SYS_IFC_ADDR			(CONFIG_SYS_IMMR + 0x00530000)
 #define CONFIG_SYS_FSL_ESDHC_ADDR		(CONFIG_SYS_IMMR + 0x00560000)
+#define CONFIG_SYS_FSL_ESDHC_1_ADDR		(CONFIG_SYS_IMMR + 0x00580000)
 #define CONFIG_SYS_FSL_CSU_ADDR			(CONFIG_SYS_IMMR + 0x00510000)
 #define CONFIG_SYS_FSL_GUTS_ADDR		(CONFIG_SYS_IMMR + 0x00ee0000)
 #define CONFIG_SYS_FSL_RST_ADDR			(CONFIG_SYS_IMMR + 0x00ee00b0)
diff --git a/include/configs/ls1012a_common.h b/include/configs/ls1012a_common.h
index 121824c..89d1370 100644
--- a/include/configs/ls1012a_common.h
+++ b/include/configs/ls1012a_common.h
@@ -94,6 +94,8 @@
 #ifdef CONFIG_MMC
 #define CONFIG_CMD_MMC
 #define CONFIG_FSL_ESDHC
+#define CONFIG_FSL_ESDHC_TWO_CONTROLLERS_SUPPORT
+#define CONFIG_FSL_ESDHC_1_NON_REMOVABLE_CARD
 #define CONFIG_SYS_FSL_MMC_HAS_CAPBLT_VS33
 #define CONFIG_GENERIC_MMC
 #define CONFIG_CMD_FAT
-- 
1.7.9.5

